### PR TITLE
refactor: use intance folder or .env to hide secrets

### LINKED ISSUE #398: change config setup to have a public config file and private one inside the instance folder

### PR DESCRIPTION
**Pull request by rehmanis**
closes #398 
- Removed the `config_demo.py` and instead we use a public `config.py` that will be overridden locally by private `config.py` inside the `instance` folder. 
- instance folder with private empty `config.py` will be created the first time the app is initialized or run.
- added `.test.prod.env` as an example to use `.env` for production incase private folders are not allowed during deployment. `.test.prod.env` is also used in the test to see if .env variables override exiting values.
- Still need to refactor `config.json` and `config_demo.json `

### CODE REVIEW

**Comment by rehmanis** [comment]
@Abdur-rahmaanJ there is something wrong with black workflow if you push again since you need to grab the changes it did.  Also was giving access denied last PR #416. Should I disable it or keep it? Another option is to go back to trigger only on push so it will only work when merged or when you create a PR.

**Comment by Abdur-rahmaanJ** [comment]
@rehmanis No idea i was supposedly reenabling it. Better disable for now!

**Comment by Abdur-rahmaanJ** [comment]
Still checking. Add .tox and .pytest_cache to .gitignore

File: shopyo/config.py [hunk 1]
```diff
@@ -3,35 +3,27 @@
 base_path = os.path.dirname(os.path.abspath(__file__))
```
Comment (Abdur-rahmaanJ) [inline shopyo/config.py:3]: Maybe just some comments stating not to touch this file but add infos in instance/config.py

Comment (rehmanis) [reply shopyo/config.py:3]: I can add that. But they are allowed to change this file as long as it is not a private change (i.e. they are okay with it being visible). Oh you merged

**Pull request merged**